### PR TITLE
Additional potion functionality

### PR DESCRIPTION
**Pull request by DaRacci**
Adding an addition object that allows plugin devs to add data to potions allows them to add for example a boolean or a simple string that they can then use to identify if the potion was applied from their plugin.

### CODE REVIEW

File: patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+-            .put(AMBIENT, ambient)
+-            .put(PARTICLES, particles)
+-            .put(ICON, icon)
+-            .build();
++                .put(TYPE, type.getId())
++                .put(DURATION, duration)
++                .put(AMPLIFIER, amplifier)
++                .put(AMBIENT, ambient)
++                .put(PARTICLES, particles)
++                .put(ICON, icon)
```
Comment (Euphillya) [inline patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch]: Restore indentation

File: patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+@@ -242,9 +280,9 @@ public class PotionEffect implements ConfigurationSerializable {
+      * @return color of this potion's particles. May be null if the potion has no particles or defined color.
+      * @deprecated color is not part of potion effects
+      */
+-     @Deprecated
+-     @Nullable
+-     @Contract("-> null")
++    @Deprecated
++    @Nullable
++    @Contract("-> null")
```
Comment (Euphillya) [inline patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch]: Restore

**Review by Euphillya — CHANGES_REQUESTED** [review]

File: patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch (hunk 1 above)
Comment (Euphillya) [inline patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch]: Restore pls

File: patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch (hunk 2 above)
Comment (Euphillya) [inline patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch]: Restore pls

**Review by Euphillya — CHANGES_REQUESTED** [review]

File: patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch (hunk 2 above)
Comment (DaRacci) [reply patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch]: my bad sorry, first time committing something

**Comment by Euphillya** [comment]
Rebuild your patch

**Comment by DaRacci** [comment]
> 
> 
> Rebuild your patch

I think i got it right this time, is it all good now?

File: patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
++
+     /**
+      * Creates a potion effect with no defined color.
+      *
+@@ -98,7 +118,7 @@ public class PotionEffect implements ConfigurationSerializable {
+      * @param map the map to deserialize from
+      */
+     public PotionEffect(@NotNull Map<String, Object> map) {
+-        this(getEffectType(map), getInt(map, DURATION), getInt(map, AMPLIFIER), getBool(map, AMBIENT, false), getBool(map, PARTICLES, true), getBool(map, ICON, getBool(map, PARTICLES, true)));
++        this(getEffectType(map), getInt(map, DURATION), getInt(map, AMPLIFIER), getBool(map, AMBIENT, false), getBool(map, PARTICLES, true), getBool(map, ICON, getBool(map, PARTICLES, true)), getObj(map, EXTRA));
```
Comment (Euphillya) [inline patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch]: add `// Purpur`

File: patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
++            .put(EXTRA, extra) // Purpur - Adding extra object
+             .build();
+     }
+ 
+@@ -188,7 +226,7 @@ public class PotionEffect implements ConfigurationSerializable {
+             return false;
+         }
+         PotionEffect that = (PotionEffect) obj;
+-        return this.type.equals(that.type) && this.ambient == that.ambient && this.amplifier == that.amplifier && this.duration == that.duration && this.particles == that.particles && this.icon == that.icon;
++        return this.type.equals(that.type) && this.ambient == that.ambient && this.amplifier == that.amplifier && this.duration == that.duration && this.particles == that.particles && this.icon == that.icon && this.extra == that.extra;
```
Comment (Euphillya) [inline patches/api/0051-The-purpose-of-this-patch-it-to-add-additional-funct.patch]: Add `// Purpur`

**Review by Euphillya — CHANGES_REQUESTED** [review]

**Review by encode42 — APPROVED** [review]

**Comment by DaRacci** [comment]
Outdated, check https://github.com/pl3xgaming/Purpur/pull/710 for the updated PR

**Pull request closed without merging**